src/app/role/[id]/page.tsx

```diff
@@ -0,0 +1,78 @@
+'use client';
+
+import React from 'react';
+import { RiVerifiedBadgeFill } from 'react-icons/ri';
+
+import BackButtonNav from '@/components/competency/BackButtonNav';
+import RoleAccordion from '@/components/competency/role/RoleAccordion';
+import { outfit, poppins } from '@/components/FontFamily';
+
+import { CompetencyType, RoleDataType } from '@/types/type';
+
+const RoleDetails = ({ params }: { params: { id: string } }) => {
+  const rolesData = localStorage.getItem('userRole');
+  let data: RoleDataType = {
+    id: 0,
+    name: '',
+    status: '',
+    description: '',
+    competencies: [],
+  };
+  if (rolesData !== null) {
+    const roles: RoleDataType[] = JSON.parse(rolesData);
+    const rolesList = roles.filter(
+      (role) => role.id.toString() === params.id.toString()
+    );
+    data = rolesList[0];
+  }
+
+  return (
+    <div className={`${outfit.className} w-full`}>
+      <BackButtonNav heading='Roles' />
+      <div className='px-[22px] pt-[75px]'>
+        <div>
+          <p
+            className={`pb-2.5 text-sm font-medium  
+          ${data?.status?.toLowerCase() === 'completed' && 'text-[#7DCC8A]'}
```

Comment:
define a memo value outside like
`const className =useMemo(()=>{
if(condition1) return 'text-[#7DCC8A]'
if(condition2) return 'text-[#7DCC8A]'
return default
},[])`
and use it like 
`className={`--your classes--  ${className}`}`